### PR TITLE
Update gem source, gem dependencies, and fix %d interpolation

### PR DESCRIPTION
**Pull request by tysonmote**
Relying on an incrementing numeric id for %d interpolation is problematic.

### CODE REVIEW

File: miniskirt.rb [hunk 1]
```diff
@@ -26,9 +26,7 @@ def build name, attrs = {}
       (m = m.is_a?(Class) ? m : m.to_s.camelize.constantize).new.tap do |r|
         attrs.symbolize_keys!.reverse_update(h).each do |k, v|
           r.send "#{k}=", case v when String # Sequence and interpolate.
-            v.sub(/%\d*d/) {|d| d % n ||= (
```
Comment (stephencelis) [inline miniskirt.rb:29]: In the spirit of keeping things short, what do you think of something like:

In the Miniskirt class:

``` ruby
@counts = Hash.new 0
```

In this method:

``` ruby
v.sub(/%\d*d/) {|d| d % n ||= @counts[name] += 1} % attrs % n
```

**Comment by stephencelis** [comment]
So, one issue with this is that if you create a model outside of Miniskirt, you could get a collision...but I'm not sure that's a problem. What do you think?

**Comment by tysonmote** [comment]
You'd have to try awfully hard to get a collision. Seems like a non-issue.

File: miniskirt.rb (hunk 1 above)
Comment (tysonmote) [inline miniskirt.rb:29]: I don't see the benefit of using a per-model hash of counts.

Comment (stephencelis) [inline miniskirt.rb:29]: Then you could merely do `@@counts = 0` and `@@counts += 1` accordingly. No need for the `incr` method.

**Comment by disbelief** [comment]
Is it necessary to drop support for both Ruby 1.8.7 and Ruby 1.9.3? Seems like quite a jump in a single PR.

**Comment by tysonmote** [comment]
Ah! I didn't realize my branch had an open pull request on it. I made the changes for our internal purposes. Sorry about that. :)

I'll close this pull request, but I'm happy to re-open if you're interested in the earlier commits.

**Pull request closed without merging**